src/6model/reprs/Decoder.c

```diff
@@ -112,11 +112,13 @@ void MVM_decoder_ensure_decoder(MVMThreadContext *tc, MVMObject *decoder, const
 static void enter_single_user(MVMThreadContext *tc, MVMDecoder *decoder) {
     if (!MVM_trycas(&(decoder->body.in_use), 0, 1))
        MVM_exception_throw_adhoc(tc, "Decoder may not be used concurrently");
+    MVM_tc_set_ex_release_mutex(tc, (uv_mutex_t *)((uintptr_t)&(decoder->body.in_use) | 1));
```

Comment:
I think it'd be cleaner to have a function `MVM_tc_set_ex_release_atomic` that takes an `AO_t` and does the `| 1` and casting. That way, we can more easily use it from other places we might need it.